File: client-android/serenada-core/src/test/java/app/serenada/core/fakes/FakeAPIClient.kt
package app.serenada.core.fakes

import app.serenada.core.network.SessionAPIClient
import app.serenada.core.network.TurnCredentials

class FakeAPIClient : SessionAPIClient {
    var turnCredentialsResult: Result<TurnCredentials> = Result.success(
        TurnCredentials(
            username = "user",
            password = "pass",
            uris = listOf("turn:turn.example.com:3478"),
            ttl = 3600
        )
    )

    val fetchTurnCredentialsCalls = mutableListOf<Pair<String, String>>()

    override fun fetchTurnCredentials(
        host: String,
        token: String,
        onResult: (Result<TurnCredentials>) -> Unit
    ) {
        fetchTurnCredentialsCalls.add(host to token)
        onResult(turnCredentialsResult)
    }
}

File: client-ios/SerenadaCore/Tests/SerenadaCoreTests/Fakes/FakeAPIClient.swift
import Foundation
@testable import SerenadaCore

final class FakeAPIClient: SessionAPIClient {
    var turnCredentialsResult: Result<TurnCredentials, Error> = .success(
        TurnCredentials(username: "user", password: "pass", uris: ["turn:turn.example.com:3478"], ttl: 3600)
    )

    private(set) var fetchTurnCredentialsCalls: [(host: String, token: String)] = []

    func fetchTurnCredentials(host: String, token: String) async throws -> TurnCredentials {
        fetchTurnCredentialsCalls.append((host: host, token: token))
        return try turnCredentialsResult.get()
    }
}
